### PR TITLE
Provide Microsoft.Bcl.Memory package readme

### PR DESCRIPTION
**Pull request by eNeRGy164**
Fill out details for the `Microsoft.Bcl.Memory` NuGet package readme.

See https://github.com/dotnet/runtime/issues/99358

### CODE REVIEW

**Comment by ViktorHofer** [comment]
cc @buyaa-n @tarekgh @stephentoub

**Review by tarekgh — APPROVED** [review]
Thanks @eNeRGy164 for providing this change. LGTM.

@buyaa-n could you please gave a quick too.

File: src/libraries/Microsoft.Bcl.Memory/src/PACKAGE.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+using System.Text;
+
+// Original data
+byte[] data = Encoding.UTF8.GetBytes("Hello World!");
+
+Span<byte> encoded = new byte[Base64Url.GetEncodedLength(data.Length)];
+Base64Url.EncodeToUtf8(data, encoded, out int _, out int bytesWritten);
+
+string encodedString = Encoding.UTF8.GetString(encoded[..bytesWritten]);
+Console.WriteLine($"Encoded: {encodedString}");
```
Comment (buyaa-n) [inline src/libraries/Microsoft.Bcl.Memory/src/PACKAGE.md:47]: There are APIs for encoding/decoding from/to chars/string, maybe use those for the encoding part:
```suggestion
byte[] data = Encoding.UTF8.GetBytes("Hello World!");
string encodedString = Base64Url.EncodeToString(data);
Console.WriteLine($"Encoded: {encodedString}");
```

File: src/libraries/Microsoft.Bcl.Memory/src/PACKAGE.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+## Additional Documentation
+
+<!-- Links to further documentation. Remove conceptual documentation if not available for the library. -->
+
+API documentation
+
+* [System.Index](https://learn.microsoft.com/dotnet/api/system.index)
+* [System.Range](https://learn.microsoft.com/en-us/dotnet/api/system.range)
+* [System.Buffers.Text](https://learn.microsoft.com/dotnet/api/system.buffers.text)
```
Comment (buyaa-n) [inline src/libraries/Microsoft.Bcl.Memory/src/PACKAGE.md:76]:
```suggestion
* [System.Buffers.Text](https://learn.microsoft.com/dotnet/api/system.buffers.text.base64url)
```

**Review by buyaa-n — APPROVED** [review]
Left a few NITs, overall LGTM, thank you @eNeRGy164

**Pull request merged**